java/memory/src/main/java/org/apache/arrow/memory/util/ArrowBufPointer.java

```diff
@@ -58,6 +70,12 @@ public void set(ArrowBuf buf, int offset, int length) {
     this.buf = buf;
     this.offset = offset;
     this.length = length;
+
+    dirty = true;
+  }
+
+  public void setHasher(ArrowBufHasher hasher) {
+    this.hasher = hasher;
```

Comment:
consider checking to make sure hash isn't null

Reply:
Done. Thanks.